### PR TITLE
Add upload service endpoints

### PR DESCRIPTION
**Pull request by mattusifer**

### CODE REVIEW

File: src/bf/api/client/mod.rs [hunk 1]
```diff
@@ -242,83 +252,173 @@ impl Blackfynn {
...
+            .map_err(|e| bf::Error::with_chain(e, "bf:request:url"));
+
+        let f = uri.and_then(move |uri| {
+            let mut req = hyper::Request::builder()
+                .method(hyper::Method::POST)
+                .uri(uri)
+                .body(chunked_file_payload)
+                .unwrap();
+
+            // If a session token exists, use it to set the "X-SESSION-ID"
```
Comment (mikeswoods) [inline src/bf/api/client/mod.rs]: Can you change this to 'set the "Authorization: Bearer" header'?

Comment (mattusifer) [reply src/bf/api/client/mod.rs]: Good catch, done

File: src/bf/api/client/progress.rs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            cb.on_update(_update)
+        }
+    }
+}
+
+/// A type representing progress updates for a multipart upload.
+#[derive(Debug, Clone, Hash)]
+pub struct ProgressUpdate {
+    part_number: usize,
+    is_multipart: bool,
```
Comment (mikeswoods) [inline src/bf/api/client/progress.rs]: I just checked and `is_multipart` isn't used in the agent, nor in any appreciable way in this library. Feel free to remove it

Comment (mattusifer) [reply src/bf/api/client/progress.rs]: nice, good catch. I'll remove that

**Pull request merged**